File: woodroid/src/main/java/me/gilo/woodroid/data/api/ProductAttributeTermAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.AttributeTerm;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;
import java.util.Map;

public interface ProductAttributeTermAPI {

    @Headers("Content-Type: application/json")
    @POST("products/attributes/{id}/terms")
    Call<AttributeTerm> create(@Path("id") int attribute_id, @Body AttributeTerm body);

    @GET("products/attributes/{id}/terms/[term_id]")
    Call<AttributeTerm> view(@Path("id") int attribute_id, @Path("term_id") int term_id);

    @GET("products/attributes/{id}/terms")
    Call<List<AttributeTerm>> list(@Path("id") int attribute_id);

    @Headers("Content-Type: application/json")
    @PUT("products/attributes/{id}/terms/[term_id]")
    Call<AttributeTerm> update(@Path("id") int attribute_id, @Path("term_id") int term_id, @Body AttributeTerm body);

    @DELETE("products/attributes/{id}/terms/[term_id]")
    Call<AttributeTerm> delete(@Path("id") int attribute_id, @Path("term_id") int term_id);

    @DELETE("products/attributes/{id}/terms/[term_id]")
    Call<AttributeTerm> delete(@Path("id") int attribute_id, @Path("term_id") int term_id, @Query("force") boolean force);

    @POST("products/attributes/batch")
    Call<String> batch(@Body AttributeTerm body);

    @GET("products/attributes/{id}/terms")
    Call<List<AttributeTerm>> filter(@Path("id") int attribute_id, @QueryMap Map<String, String> filter);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/ProductAttributeTermAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.AttributeTerm
import retrofit2.Call
import retrofit2.http.*

interface ProductAttributeTermAPI {

    @Headers("Content-Type: application/json")
    @POST("products/productAttributes/{id}/terms")
    fun create(@Path("id") attribute_id: Int, @Body body: AttributeTerm): Call<AttributeTerm>

    @GET("products/productAttributes/{id}/terms/[term_id]")
    fun view(@Path("id") attribute_id: Int, @Path("term_id") term_id: Int): Call<AttributeTerm>

    @GET("products/productAttributes/{id}/terms")
    fun list(@Path("id") attribute_id: Int): Call<List<AttributeTerm>>

    @Headers("Content-Type: application/json")
    @PUT("products/productAttributes/{id}/terms/[term_id]")
    fun update(@Path("id") attribute_id: Int, @Path("term_id") term_id: Int, @Body body: AttributeTerm): Call<AttributeTerm>

    @DELETE("products/productAttributes/{id}/terms/[term_id]")
    fun delete(@Path("id") attribute_id: Int, @Path("term_id") term_id: Int): Call<AttributeTerm>

    @DELETE("products/productAttributes/{id}/terms/[term_id]")
    fun delete(@Path("id") attribute_id: Int, @Path("term_id") term_id: Int, @Query("force") force: Boolean): Call<AttributeTerm>

    @POST("products/productAttributes/batch")
    fun batch(@Body body: AttributeTerm): Call<String>

    @GET("products/productAttributes/{id}/terms")
    fun filter(@Path("id") attribute_id: Int, @QueryMap filter: Map<String, String>): Call<List<AttributeTerm>>

}